When zone is deleted from route53 ns_propagated check fails

https://sentry.presslabs.net/presslabs/zinc/issues/475/

```
AssertionError: 
(1 additional frame(s) were not displayed)
...
  File "celery/app/trace.py", line 622, in __protected_call__
    return self.run(*args, **kwargs)
  File "zinc/tasks.py", line 56, in update_ns_propagated
    models.Zone.update_ns_propagated()
  File "zinc/models.py", line 347, in update_ns_propagated
    zone.ns_propagated = ns_check.is_ns_propagated(zone, resolver=resolver)
  File "zinc/ns_check.py", line 20, in is_ns_propagated
    r53_name_servers = set(zone.route53_zone.ns['values'])
  File "zinc/route53.py", line 128, in ns
    assert len(ns) == 1

AssertionError
```

--- a/zinc/ns_check.py
+++ b/zinc/ns_check.py
@@ -15,6 +15,8 @@ def get_resolver():
 
 
 def is_ns_propagated(zone, resolver=None):
+    if not zone.route53_zone.exists:
+        return False
     if resolver is None:
         resolver = get_resolver()
     r53_name_servers = set(zone.route53_zone.ns['values'])

--- a/zinc/route53.py
+++ b/zinc/route53.py
@@ -144,6 +144,8 @@ class Zone(object):
 
     @property
     def ns(self):
+        if not self.exists:
+            return None
         ns = [record for record in self.records().values()
               if record['type'] == 'NS' and record['name'] == '@']
         assert len(ns) == 1

